src/hex/shape.rs

```diff
@@ -51,16 +73,176 @@ impl Shape<Axial> {
         }
     }
 
+    pub fn resize(&mut self, scale: (f64, f64)) -> &Self {
+        todo!()
+    }
+
+    // Bake the transform into localspace
+    pub fn bake_transform(&mut self) -> HexShape<T> {
+        todo!()
+    }
+
     // Retrieves the hex array in local space
-    pub fn get_hexes(&self) -> Vec<Axial> {
-        self.tiles.clone()
+    pub fn get_hexes(&self) -> &[HexTile<T>] {
+        self.tiles.as_slice()
     }
 
     // Retrieves the hex array in grid space
-    pub fn get_hexes_gs(&self) -> Vec<Axial> {
+    // Because we modify the HexTile coordinates into gs we must
+    // return a cloned vector instead of a slice.
+    pub fn get_hexes_gs(&mut self) -> Vec<HexTile<T>>
+    where
+        T: Clone
+    {
         self.tiles
             .iter()
-            .map(|&coord| coord.apply_all(self.transform))
+            .map(|tile| {tile.coord.apply_all(self.transform); tile.clone()})
             .collect()
     }
 }
+
+// General shape constructors
+// Move this to another file? Inside a different module?
+pub mod shape_builders {
+    use crate::{axial, transform};
+
+    use super::*;
+
+    pub struct Inequality {
+        q_min: i32,
+        q_max: i32,
+        r_min: i32,
+        r_max: i32,
+        s_min: i32,
+        s_max: i32,
+    }
+
+    impl Inequality {
```

Comment:
Perhaps could use a short comment explaining what the purpose of this is and why it's named `Inequality` as it's not immediately intuitive to me.

Reply:
Provided a comment on the Inequality struct. Essentially it solves a system of linear inequalities in order to facilitate constructing shapes given vertices.